test/puppetlabs/puppetdb/cli/import_export_roundtrip_test.clj

```diff
@@ -153,7 +179,32 @@
     (jutils/puppetdb-instance
      (assoc-in (jutils/create-config) [:command-processing :max-frame-size] "1024")
      (fn []
-       (is (empty? (export/get-nodes "localhost" jutils/*port*)))
-       (submit-command :replace-catalog 5 catalog)
-       (is (thrown-with-msg? java.util.concurrent.ExecutionException #"Results not found"
-                             @(block-until-results 5 (json/parse-string (export/catalog-for-node "localhost" jutils/*port*  "foo.local")))))))))
+       (is (empty? (export/get-nodes *base-url*)))
+       (submit-command *base-url* :replace-catalog 5 catalog)
+       (is (thrown-with-msg?
+            java.util.concurrent.ExecutionException #"Results not found"
+            @(block-until-results 5
+                                  (json/parse-string
+                                   (export/catalog-for-node *base-url*
+                                                            "foo.local")))))))))
+
+(defn- check-invalid-url-handling [cmd expected-msg-re]
+  (is (re-find expected-msg-re
+               (with-out-str
```

Comment:
I think [this](https://github.com/puppetlabs/puppetdb/blob/master/test/puppetlabs/puppetdb/testutils.clj#L291) would simplify the code here.